provider/datagen/src/lib.rs

```diff
@@ -338,6 +340,7 @@ pub enum Out {
     },
 }
 
+#[deprecated]
```

Comment:
Suggestion: Let's keep the current one stable and make the new one `#[cfg(feature = "experimental")]` so that we can check it in and iterate